gcs/src/main/scala/monix/connect/gcs/utiltiies/StorageDownloader.scala

```diff
@@ -0,0 +1,47 @@
+package monix.connect.gcs.utiltiies
+
+import java.io.FileOutputStream
+import java.nio.channels.Channels
+import java.nio.file.Path
+
+import cats.effect.Resource
+import com.google.cloud.ReadChannel
+import com.google.cloud.storage.{BlobId, Storage => GoogleStorage}
+import monix.eval.Task
+import monix.reactive.Observable
+
+trait StorageDownloader {
+
+  private def openReadChannel(storage: GoogleStorage, bucket: String, blobId: BlobId): Resource[Task, ReadChannel] = {
+    Resource.make {
+      Task(storage.reader(bucket, blobId.getName))
+    } { reader =>
+      Task(reader.close())
+    }
+  }
+
+  private def openOutputStream(path: Path): Resource[Task, FileOutputStream] = {
+    Resource.make {
+      Task(new FileOutputStream(path.toFile))
+    } { os =>
+      Task(os.close())
+    }
+  }
+
+  private def download(reader: ReadChannel, os: FileOutputStream, chunkSize: Int): Task[Unit] = {
+    Observable
+      .fromInputStreamUnsafe(Channels.newInputStream(reader), chunkSize)
+      .takeWhile(_.nonEmpty)
+      .mapEval(bytes => Task(os.write(bytes)))
+      .completedL
+  }
+
+  protected def downloadFromBucket(storage: GoogleStorage, bucket: String, blobId: BlobId, path: Path, chunkSize: Int): Task[Unit] = {
```

Comment:
I like this implementation, however (correct me if I am wrong), but I think it would be more adequate renaming it to something like `downloadAndSave`, `downloadToLocalFS` or similar? Making sure that the user know by hand that the content of the object will end up to the specified path in the local file system. (or maybe it's ok)

Therefore, apart of this method I would define another one that does not depend on the cats `Resource` data type.
Therefore this implementation would read the google object as an `Observable[Array[Byte]]` that the user could use as they want to, so they won't stick to save the data locally. That one could be the one called `download()`  or `read`. (I mean to do something similar to what I did for the parquet reader at `package monix.connect.parquet.ParquetPublisher`

Reply:
Maybe that second method would just be better to implement with the `Resumable` interface that you mentioned in the description...

Reply:
I agree, I will add the additional methods, and rename the existing ones to their behavior is more explicit.